pkg/metrics/util.go

```diff
@@ -22,3 +23,33 @@ func EventPayload(ev types.Event) ([]byte, error) {
 	payload := map[string]interface{}{"event": ev, "versions": vmap}
 	return json.Marshal(payload)
 }
+
+var secretKeywords = []string{"password", "secret", "token", "key"}
+
+// redactFlags redacts presumed secret values from a string slice based on preset keywords.
+// For example, given ["--license", "./path", "--password", "123"], it will return ["--license", "./path", "--password", "*****"]
+func redactFlags(flags []string) []string {
+	result := make([]string, len(flags))
+
+	valueHasSecret := false
+	for i, flag := range flags {
+		// Check if the is the flag name or the value
+		isFlagKey := strings.HasPrefix(flag, "-")
```

Comment:
If the string we want to redact (like the password or the key) starts with a dash, we would miss redacting it. This is probably a very unlikely edge case, but it could still occur.

Reply:
Good call, let me check for that.